### DISCUSSION TITLE
OpenJDK configure detects WSL2 as WSL1

### DISCUSSION THREAD

**Discussion by evilensky** [Homebrew/discussions#1646] [Linux] [answered]
I'm not sure where to discuss my issue but it seems that OpenDJK autoconf is detecting WSL2 as WSL1...i don't think this is good behavior since ideally it treats Debian in WSL2 as a Linux...forgive me if I am ignorant why OpenJDK wants to know about WSL1 in the first place...

Google brought me this upstream issue: https://bugs.openjdk.java.net/browse/JDK-8226521

Can anyone share any tips, can I patch out that detection of WSL2 so that the configure scripts treats my WSL2 distro simply as a a Linux distro?

```
==> Installing dependencies for rubberband: openjdk
==> Installing rubberband dependency: openjdk
==> ./configure --disable-warnings-as-errors --with-boot-jdk-jvmargs=-Duser.home=/home/evilensk/.cache/Homebrew/java_cache --with-boot-jdk=/tmp/openjdk-20210608-26456-1
Last 15 lines from /home/evilensk/.cache/Homebrew/Logs/openjdk/01.configure:
checking Windows environment type... wsl1
checking wsl1 drive prefix... '/mnt'
checking wsl1 root directory as Windows path... '[unavailable]'
checking wsl1 temp directory... /bin/wslpath: Invalid argument
Usage:    -a    force result to absolute path format    -u    translate from a Windows path to a WSL path (default)    -w    translate from a WSL path to a Windows path    -m    translate from a WSL path to a Windows path, with '/' instead of '\'EX: wslpath 'c:\users'
checking wsl1 release... 5.4.72-microsoft-standard-WSL2
checking wsl1 version... #1 SMP Wed Oct 28 23:40:43 UTC 2020
checking Windows version... /tmp/openjdk-20210608-26456-17hrmks/jdk16u-jdk-16.0.1-ga/build/.configure-support/generated-configure.sh: line 34632: cd: too many arguments

checking wsl distribution... /tmp/openjdk-20210608-26456-17hrmks/jdk16u-jdk-16.0.1-ga/build/.configure-support/generated-configure.sh: line 34657: -d: command not found

/bin/wslpath: : Invalid argument
/bin/wslpath: : Invalid argument
configure: error: The path of FIXPATH_DIR, which resolves as "/tmp/openjdk-20210608-26456-17hrmks/jdk16u-jdk-16.0.1-ga/make/scripts", could not be imported.
configure exiting with result code 1
                                                                                                                        
```

**Comment by danielnachun** [comment, marked as answer]
We have a PR open for this right now: https://github.com/Homebrew/linuxbrew-core/pull/22620.

On a related note, if you are using WSL, I'd strongly recommend using the default prefix of `/home/linuxbrew/.linuxbrew` for Homebrew since you should have `sudo` access.  This will allow you to use all of our Linux bottles, not just the ones with `cellar :any`.  This would save you the trouble of having to build OpenJDK from source.

**Reply by evilensky** [reply]
Much obliged! 

Yes I should probably re install with the default prefix. Maybe when the next Debian is released…
